src/mlpack/methods/decision_tree/decision_tree.hpp

```diff
@@ -371,6 +409,12 @@ class DecisionTree :
    */
   arma::vec classProbabilities;
 
+  // this stores the starting point in the dataset that belongs to this node.
+  size_t beginIndex;
+
+  // this stores the count of the data points stored in this node.
+  size_t countInNode;
```

Comment:
It's really important that we keep the `DecisionTree` object as small as possible, so since these are only used for training, I'd suggest that we find a way to move them into some auxiliary training structure that isn't stored in the tree.

Reply:
@rcurtin 
By addition of two variables (countinNode and beginIndex) the problem of finding the majority class in the node is solved. I understand that it will increase the size of the class and hence the overheads for building the tree and so I will think of storing them so that the overheads aren't increased.